src/main/kotlin/com/manimdsl/frontend/SemanticAnalysis.kt

```diff
@@ -61,7 +62,7 @@ class SemanticAnalysis {
 
         return when (expression) {
             is AddExpression, is SubtractExpression, is MultiplyExpression -> {
-                if (expr1Type is NumberType && expr2Type is NumberType) NumberType else ErrorType
+                if ((expr1Type is NumberType || expr1Type is CharType) && (expr2Type is NumberType || expr2Type is CharType)) NumberType else ErrorType
```

Comment:
Might be good to have a "Comparable" interface or something so that both number and char are in one go

Reply:
Also if we're allowing these kind of expressions where left is char and right is number, does that mean you can do let x:int = 'a'; ?

Reply:
If that's not possible, it's a bit confusing, as it means if I just want the number representation of a character, I have to do let x: int = 'a' * 1; ?

I might be missing something though

Reply:
Yeah that would be how it is. In java it is like that but java has casting. It generally covers most cases I would say

Reply:
Just to clarify we are not allowing assignments `x: number = 'a';`

Reply:
Okay cool, should we add a toInt('a') and toChar(97) or no?

Reply:
Sure

Reply:
will toInt('3') return 3 or its ascii